leader-election-impl/src/main/java/com/palantir/leader/proxy/LeadershipStateManager.java

```diff
@@ -135,6 +135,14 @@ private void closeDelegate(Object delegate) {
         }
     }
 
+    void handleDelegateNoLongerValid() {
+        // This forces a re-creation of the delegate on the next call to getOrUpdateLeadershipToken. We cannot just
+        // clear the delegate reference and/or invoke the supplier here - doing the former will result in this not
```

Comment:
I was trying to reject three "obvious" but wrong implementations of this method:
- `clearDelegate()`: this will not actually ever reset the delegate.
- `delegateRef.set(delegateSupplier.get())`: this creates a race condition as the supplier might update the delegate underneath us and we might accidentally clear it
- `delegateRef.compareAndSet(delegateRef.get(), delegateSupplier.get())`: this creates a race condition where the supplier creates a new delegate that races with us

I'll clean this up so it reads better.